### ISSUE TITLE
Update package versions all environment.ymls

### ISSUE THREAD

**Issue by grallewellyn** [MAAP-Project/ZenHub#915]
https://github.com/MAAP-Project/maap-py/issues/76#issuecomment-1915241990
Bugs version should be 0 (ie x.x.0) and using flexible versioning like (~=)
Also want to do this for R binaries because it started failing to build in late February

**Comment by chuckwondo** [comment]
@grallewellyn, just to be clear, when you use compatible release constraints, you typically want to specify only the major and minor version numbers, not the patch/bug version number.

For example, to constrain a package to major version number `1`, you generally want to use `~=1.X`, which is equivalent to `>=1.X,==1.*`, where `X` can be some minimum minor version number you want, which may simply be `0`. You would typically set `X` to some value greater than `0` when you want to make use of some feature introduced in some particular minor version.

For example, if the package introduced a feature you want to use in version `1.4`, you would use `~=1.4` to make sure you get _at least_ version `1.4`, but will take any version `>=1.4` _and_ `==1.*`. This ensures that you get _at least_ the version containing the feature you want to use (and perhaps subsequent features and bug fixes), but _don't_ get anything from a later _major_ version, which might contain breaking changes.

You _can_ use `~=1.X.Y` (equivalent to `>=1.X.Y,==1.X.*`), but that will fix both the major _and_ minor version numbers, letting _only_ the patch/bug version (`Y`) vary. In most cases, this is _too_ restrictive, reducing flexibility for dependency resolution, increasing the likelihood of resolution failure, and should be avoided unless you are somehow forced to constrain both the major and minor version numbers for some reason, such as some other transitive dependency constraint.

See [PEP 440's compatible release section](https://peps.python.org/pep-0440/#compatible-release) for more information and examples.

**Comment by grallewellyn** [comment]
@chuckwondo Thank you for this information! 
I wanted to pin both the major and minor versions because of what was happening in maap-py with boto3, but if this is not the best practice, then we can just pin the major version and potentially set the minor version to be greater than a certain version if needed

**Comment by chuckwondo** [comment]
Right, with the maap-py boto3 depenency, that was one of those rare cases where we were forced to pin the minor version (`~=1.33.0`) in order for a successful dependency resolution. IIRC, that's perhaps because other dependency constraints on dependencies in the list that included maap-py were too tight, so loosening the other constraints a bit (i.e., to `~=X.Y`) should allow us to subsequently loosen `maap-py`'s boto3 constraint to `~=1.33`, so we're not "stuck" with some `1.33.X` version.

**Comment by grallewellyn** [comment]
After the whole issue with DIT failing to launch all new images due to jupyter_server being upgraded from 2.12.5 to 2.13.0 (we only had the major version pinned) and stopping development for a couple days, we have decided to pin the major and minor version when we implement this ticket

**Comment by gchang** [comment]
We could try doing a snapshot of all current libraries at the beginning/mid cycle of a milestone. This allows us to stay up to date at a reasonable cadence. @grallewellyn to update release process to highlight this.

**Comment by grallewellyn** [comment]
Updated the release process. I also modified the ticket title to reflect that we are no longer doing flexible versions for the environment.ymls since we don't have any testing in place currently to ensure that the package upgrades don't cause bugs 
Instead we are going to do a `conda update --all` at the beginning of a milestone so that we can test the updates throughout the release cycle
